### PR TITLE
[FRONT-745] Fix StreamrClient creation in StreamrClientProvider

### PR DESCRIPTION
**Pull request by juhah**
Issue was caused by the session token not being updated since it was being fetched statically. I changed the config to update every time the session token updates. I also cleaned the multiple session contexts and client provider contexts since there is always one present at the root level.

### CODE REVIEW

File: app/src/auth/components/LoginPage.jsx [hunk 1]
```diff
@@ -204,7 +203,5 @@ const LoginPage = () => {
 export { LoginPage }
 
 export default userIsNotAuthenticated((props) => (
-    <SessionProvider>
```
Comment (juhah) [inline app/src/auth/components/LoginPage.jsx:207]: This is already present at the root level.

File: app/src/auth/components/LogoutPage.jsx [hunk 2]
```diff
@@ -25,6 +27,7 @@ const LogoutPage = () => {
             })
             if (isMounted()) {
                 dispatch(logoutAction())
+                resetSessionToken()
```
Comment (juhah) [inline app/src/auth/components/LogoutPage.jsx:30]: Maybe the cleanup can be combined somehow, now added to clear session context here as well

File: app/src/marketplace/containers/GlobalInfoWatcher/index.jsx [hunk 3]
```diff
@@ -165,7 +167,8 @@ export const GlobalInfoWatcher = ({ children }: Props) => {
     const onContinue = useCallback(() => {
         setAccountChanged(false)
         dispatch(logout())
-    }, [dispatch])
+        resetSessionToken()
```
Comment (juhah) [inline app/src/marketplace/containers/GlobalInfoWatcher/index.jsx:170]: Clear session context when switching accounts

File: app/src/shared/components/StreamrClientProvider.jsx [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+        const nextConfig = getClientConfig()
+
+        if (token) {
+            nextConfig.auth = {
+                sessionToken: token,
+            }
+        }
+
+        return nextConfig
+    }, [token])
```
Comment (juhah) [inline app/src/shared/components/StreamrClientProvider.jsx:19]: Config never updated after first call when using only `getClientConfig`

**Review by fonty1 — APPROVED** [review]
LGTM

**Pull request merged**